modules/dnn/src/dnn.cpp

```diff
@@ -3187,19 +3180,19 @@ struct Net::Impl : public detail::NetImplBase
         CV_LOG_DEBUG(NULL, "updateLayersShapes() - DONE");
     }
 
-    LayerPin getLatestLayerPin(const std::vector<LayerPin>& pins)
+    LayerPin getLatestLayerPin(const std::vector<LayerPin>& pins) const
     {
         return *std::max_element(pins.begin(), pins.end());
     }
 
-    Mat getBlob(const LayerPin& pin)
+    Mat getBlob(const LayerPin& pin) const
     {
         CV_TRACE_FUNCTION();
 
         if (!pin.valid())
             CV_Error(Error::StsObjectNotFound, "Requested blob not found");
 
-        LayerData &ld = layers[pin.lid];
+        const LayerData &ld = const_cast<MapIdToLayerData&>(layers)[pin.lid];
         if ((size_t)pin.oid >= ld.outputBlobs.size())
```

Comment:
If we create `[pin.lid]` in layers in previous line, its `outputBlobs.size()` is zero and the condition is true, so we can get rid of const_cast.

Reply:
Updated